src/rideables/MontageGraph.hpp

```diff
@@ -221,348 +290,405 @@ class MontageGraph : public RGraph, public Recoverable{
                         }
-                    }
-                    for (Relation *r : toRemoveList) {
-                        _v->adjacency_list.erase(r);
-                        garbageList.push_back(r);
-                    }
-                    toRemoveList.clear();
-
-                    for (Relation *r : _v->dest_list) {
-                        if (r->get_unsafe_dest(this) == id) {
-                            toRemoveList.push_back(r);
+                        for (auto r : destination(other)) {
+                            if (r->src() == vid) {
+                                std::cout << "Them: (" << r->src() << "," << r->dest() << ")" << std::endl;
+                            }
                         }
+                        for (auto r : destination(vid)) {
+                            if (r->src() == other) {
+                                std::cout << "Us: (" << r->src() << "," << r->dest() << ")" << std::endl;
+                            }
+                        }
+                        for (auto r : source(other)) {
+                            if (r->dest() == vid) {
+                                std::cout << "Them: (" << r->src() << "," << r->dest() << ")" << std::endl;
+                            }
+                        }
+                        std::abort();
                     }
-                    for (Relation *r : toRemoveList) {
-                        _v->dest_list.erase(r);
-                        garbageList.push_back(r);
-                    }
-                }
+                    
+                    remove_relation(source(other), &src);
+                    remove_relation(destination(other), &dest);
+                    assert(!has_relation(source(other), &src) && !has_relation(destination(other), &dest));
+                }                
                 
-                // Step 4: Delete edges, clear set of src and dest edges
-                v->adjacency_list.clear();
-                v->dest_list.clear();
-                {
-                    MontageOpHolder _holder(this);
-                    for (Relation *r : garbageList) {
-                        pdelete(r);
-                    }
-                }
+                std::vector<Relation*> toDelete(source(vid).size() + destination(vid).size());
+                for (auto r : source(vid)) toDelete.push_back(r);
+                for (auto r : destination(vid)) toDelete.push_back(r);
+                source(vid).clear();
+                destination(vid).clear();
+                destroy(vid);
+                for (auto r : toDelete) if (r != nullptr) pdelete(r);
                 
-                // Step 5: Release in reverse order
-                std::reverse(vertices.begin(), vertices.end());
-                for (int _vid : vertices) {
-                    idxToVertex[_vid]->seqNumber++;
-                    idxToVertex[_vid]->unlock();
+                // Step 4: Release in reverse order
```

Comment:
`end_op` before releasing locks; let's have a pair of curly braces from line 579 to 620